repository_service_tuf_worker/repository.py

```diff
@@ -203,6 +204,21 @@ def refresh_settings(self, worker_settings: Optional[Dynaconf] = None):
         self._worker_settings = settings
         return settings
 
+    def write_repository_settings(self, key: str, value: Any):
+        """
+        Writes repository settings.
+
+        Repository settings uses Dynaconf and it is persistent in the Redis
+        server. It stores as a dictionary with key and value.
+
+        Args:
+            key: key name
+            value: value for the key
+        """
+        settings_data = self._settings.as_dict(env=self._settings.current_env)
+        settings_data[key] = value
```

Comment:
By changing the settings data don't you also need to save it back in the `Dynaconf` `self._settings` object?

Reply:
The `self._settings.get_fresh()` [manages to force load from the dynaconf store](https://dynaconf.readthedocs.io/en/docs_223/reference/dynaconf.html#dynaconf.base.Settings.fresh).

I did not understand your question here—what do you mean by calling the `save` function?
The flow for save settings is as follows:
`bootstrap()` receives settings as a dict and calls `save_settings` (which prepares the keys/values) and calls `write_repository_settings()`, [which basics use the Dynaconf `redis_loader.write` to persist it on Redis](https://dynaconf.readthedocs.io/en/docs_223/reference/dynaconf.loaders.html#dynaconf.loaders.redis_loader.write).
The same will be done in the future by `metadata_rotate`, which basically will overwrite this settings.

Reply:
After a conversation with @kairoaraujo I realized that `redis_loader` below is an API call from Dynaconf and given that the `self_settings` Dynaconf object uses `Redis` as an internal storage it will make sure that it writes this new setting in the `Redis` server and at `Dynaconf` object as well.